src/prss.rs

```diff
@@ -0,0 +1,493 @@
+use crate::field::Field;
+use aes::{
+    cipher::{generic_array::GenericArray, BlockEncrypt, KeyInit},
+    Aes256,
+};
+use byteorder::{ByteOrder, NativeEndian};
+use hkdf::Hkdf;
+use rand::{CryptoRng, RngCore};
+use sha2::Sha256;
+use x25519_dalek::{EphemeralSecret, PublicKey};
+
+/// A participant in a 2-of-3 replicated secret sharing.
+pub struct Participant {
+    left: Generator,
+    left_bits: BitGenerator,
+    right: Generator,
+    right_bits: BitGenerator,
+}
+
+impl Participant {
+    /// Generate two random values, one that is known to the left helper
+    /// and one that is known to the right helper.
+    #[must_use]
+    pub fn generate_values(&self, index: u128) -> (u128, u128) {
+        (self.left.generate(index), self.right.generate(index))
+    }
+
+    /// Generate two random field values, one that is known to the left helper
+    /// and one that is known to the right helper.
+    #[must_use]
+    pub fn generate_fields<F: Field>(&self, index: u128) -> (F, F) {
+        let (l, r) = self.generate_values(index);
+        (F::from(l), F::from(r))
+    }
+
+    /// Generate an additive share of zero.
```

Comment:
This one is subtle. probably merits a comment explaining how it works. Took me a second to work it out =).